drivers/dma/dma_dw_common.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/dma.h>
+#include <soc.h>
+#include "dma_dw_common.h"
+
+#define LOG_LEVEL CONFIG_DMA_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dma_dw_common);
+
+#define BYTE				(1)
+#define WORD				(2)
+#define DWORD				(4)
+
+/* CFG_LO */
+#define DW_CFG_CLASS(x)			(x << 29)
+/* CFG_HI */
+#define DW_CFGH_SRC_PER(x)		((x & 0xf) | ((x & 0x30) << 24))
+#define DW_CFGH_DST_PER(x)		(((x & 0xf) << 4) | ((x & 0x30) << 26))
+
+/* default initial setup register values */
+#define DW_CFG_LOW_DEF			0x0
+
+
+/* number of tries to wait for reset */
+#define DW_DMA_CFG_TRIES	10000
+#define INT_MASK_ALL		0xFF00
+
+static ALWAYS_INLINE void dw_write(uint32_t dma_base, uint32_t reg, uint32_t value)
+{
+	*((volatile uint32_t*)(dma_base + reg)) = value;
+}
+
+static ALWAYS_INLINE uint32_t dw_read(uint32_t dma_base, uint32_t reg)
+{
+	return *((volatile uint32_t*)(dma_base + reg));
+}
+
+void dw_dma_isr(const struct device *dev)
+{
+	const struct dw_dma_dev_cfg *const dev_cfg = dev->config;
+	struct dw_dma_dev_data *const dev_data = dev->data;
+	struct dma_chan_data *chan_data;
+
+	uint32_t status_tfr = 0U;
+	uint32_t status_block = 0U;
+	uint32_t status_err = 0U;
```

Comment:
Not added in this patch, but: unnecessary init of "status_err", it's initialized on L62.